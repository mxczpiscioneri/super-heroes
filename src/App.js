import React from 'react'
import { StatusBar } from 'react-native'
import Routes from './routes'
import Colors from './statics/colors'
import FavoriteContext from './components/Favorite/context'

const App = () => (
  <FavoriteContext.Provider>
    <StatusBar barStyle="light-content" backgroundColor={Colors.SECONDARY} />
    <Routes />
  </FavoriteContext.Provider>
)

export default App
