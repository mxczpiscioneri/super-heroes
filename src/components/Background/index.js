import React from 'react'
import PropTypes from 'prop-types'
import { ImageBackground } from './styles'
import { BackgroundImage } from '../../statics/images'

export const Background = props => (
  <ImageBackground source={BackgroundImage}>
    {props.children}
  </ImageBackground>
)

Background.propTypes = {
  children: PropTypes.any,
}

Background.defaultprops = {
  children: '',
}
