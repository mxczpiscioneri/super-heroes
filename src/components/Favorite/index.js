import React from 'react'
import PropTypes from 'prop-types'
import {
  Icon,
  Image,
} from './styles'
import {
  IconFavorite,
  IconFavoriteBorder,
} from '../../statics/images'

export const Favorite = props => {
  const { character, favoritesIds, setFavorite } = props
  const isFavorite = favoritesIds.indexOf(character.id) >= 0

  return (
    <Icon activeOpacity={0.6} onPress={() => setFavorite(character)}>
      <Image source={isFavorite ? IconFavorite : IconFavoriteBorder} />
    </Icon>
  )
}

Favorite.propTypes = {
  character: PropTypes.object.isRequired,
  favoritesIds: PropTypes.array.isRequired,
  setFavorite: PropTypes.func.isRequired,
}
