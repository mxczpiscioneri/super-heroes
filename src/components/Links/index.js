import React from 'react'
import PropTypes from 'prop-types'
import { Linking } from 'react-native'
import {
  Container,
  Link,
  Text,
} from './styles'

export const Links = props => (
  <Container>
    {props.urls.map(item => (
      <Link
        key={item.type}
        activeOpacity={0.8}
        onPress={() => { Linking.openURL(item.url) }}
      >
        <Text>{item.type}</Text>
      </Link>
    ))}
  </Container>
)

Links.propTypes = {
  urls: PropTypes.array.isRequired,
}
