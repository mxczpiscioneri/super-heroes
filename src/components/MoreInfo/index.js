import React from 'react'
import PropTypes from 'prop-types'
import Carousel from 'react-native-snap-carousel'
import { Dimensions } from 'react-native'
import { generateUrlImage } from '../../utils'
import {
  Container,
  Title,
  Box,
  Image,
  Legend,
} from './styles'

const { width } = Dimensions.get('window')

export const MoreInfo = props => (
  <Container>
    <Title>{props.title}</Title>

    <Carousel
      data={props.data}
      sliderWidth={width}
      itemWidth={166}
      renderItem={({ item }) => (
        <Box key={item.id}>
          {item.thumbnail &&
            <Image source={{ uri: generateUrlImage(item.thumbnail.path, item.thumbnail.extension, 'portrait_xlarge') }} />
          }
          <Legend>{item.title}</Legend>
        </Box>
      )}
    />
  </Container>
)

MoreInfo.propTypes = {
  title: PropTypes.string.isRequired,
  data: PropTypes.array.isRequired,
}
