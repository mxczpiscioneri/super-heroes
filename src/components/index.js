export * from './Background'
export * from './Character'
export * from './Favorite'
export * from './Links'
export * from './MoreInfo'
