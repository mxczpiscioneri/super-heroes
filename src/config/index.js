import RNConfig from 'react-native-config'

const Config = {
  ENV: RNConfig.ENV,
  API_URL: RNConfig.API_URL,
  API_KEY_PUBLIC: RNConfig.API_KEY_PUBLIC,
  API_KEY_PRIVATE: RNConfig.API_KEY_PRIVATE,
  API_TIMEOUT: parseInt(RNConfig.API_TIMEOUT) || 60000,
}

export default Config
