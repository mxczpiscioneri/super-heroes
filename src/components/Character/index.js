import React from 'react'
import PropTypes from 'prop-types'
import FavoriteContext from '../Favorite/context'
import {
  Container,
  Box,
  Image,
  Legend,
  FavoriteBox,
} from './styles'
import { Favorite } from '../Favorite'

export const Character = props => {
  const {
    character,
    image,
    onPress,
    legend,
  } = props

  return (
    <Container>
      <Box
        activeOpacity={0.9}
        onPress={onPress}
      >
        <Image source={{ uri: image }} />
        {legend &&
          <Legend>{character.name}</Legend>
        }
      </Box>
      <FavoriteBox>
        <FavoriteContext.Consumer>
          {context => (
            <Favorite {...context} character={character} />
          )}
        </FavoriteContext.Consumer>
      </FavoriteBox>
    </Container>
  )
}

Character.propTypes = {
  character: PropTypes.object.isRequired,
  image: PropTypes.string.isRequired,
  onPress: PropTypes.func,
  legend: PropTypes.bool,
}

Character.defaultprops = {
  onPress: null,
  legend: '',
}
