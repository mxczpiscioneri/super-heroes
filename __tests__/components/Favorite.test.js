import React from 'react'
import renderer from 'react-test-renderer'
import { AsyncStorage } from 'react-native'
import { shallow } from 'enzyme'
import { Favorite } from '../../src/components'
import { characterMock } from '../../__mocks__/characters'
import {
  IconFavorite,
  IconFavoriteBorder,
} from '../../src/statics/images'

describe('Favorite', () => {
  test('Favorite snapshot', () => {
    const rendered = renderer.create(
      <Favorite
        character={characterMock}
        favorites={[]}
        favoritesIds={[]}
        setFavorite={jest.fn()}
      />,
    ).toJSON()
    expect(rendered).toMatchSnapshot()
  })

  test('Favorite renders correctly', () => {
    const wrapper = shallow(
      <Favorite
        character={characterMock}
        favorites={[]}
        favoritesIds={[]}
        setFavorite={jest.fn()}
      />
    )

    expect(wrapper.length).toEqual(1)
    expect(wrapper.find('Styled(TouchableOpacity)').length).toEqual(1)
    expect(wrapper.find('Styled(Image)').length).toEqual(1)
    expect(wrapper.find('Styled(Image)').props().source).toEqual(IconFavoriteBorder)
  })

  test('Favorite renders correctly active', async () => {
    await AsyncStorage.setItem('favorites', JSON.stringify([characterMock]))

    const wrapper = shallow(
      <Favorite
        character={characterMock}
        favoritesIds={[characterMock.id]}
        setFavorite={jest.fn()}
      />
    )

    expect(wrapper.length).toEqual(1)
    expect(wrapper.find('Styled(TouchableOpacity)').length).toEqual(1)
    expect(wrapper.find('Styled(Image)').length).toEqual(1)
    expect(wrapper.find('Styled(Image)').props().source).toEqual(IconFavorite)
  })
})
