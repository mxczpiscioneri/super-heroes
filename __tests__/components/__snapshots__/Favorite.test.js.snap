// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite Favorite snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/statics/images/icon_favorite_border.png",
      }
    }
    style={
      Array [
        Object {},
      ]
    }
  />
</View>
`;
